src/preact/base-element.js

```diff
@@ -153,9 +153,28 @@ export class PreactBaseElement extends AMP.BaseElement {
       notify: () => this.mutateElement(() => {}),
     };
 
+    /** @private {?API_TYPE} */
+    this.apiWrapper_ = null;
+
     /** @private {{current: ?API_TYPE}} */
     this.ref_ = createRef();
 
+    /** @param {?API_TYPE|null} current */
+    this.refSetter_ = (current) => {
+      // The API shape **must** be consistent.
+      if (current !== null) {
+        if (this.apiWrapper_) {
+          this.checkApiWrapper_(current);
+        } else {
+          this.initApiWrapper_(current);
+        }
+      }
+      this.ref_.current = current;
```

Comment:
Nit, but in this case we don't need to use `createRef` at all since it's all local to our code now - a simple var would do.